contrib/chromium/template.py

```diff
@@ -1,45 +1,49 @@
 pkgname = "chromium"
 # https://chromiumdash.appspot.com/releases?platform=Linux
-pkgver = "120.0.6099.224"
+pkgver = "121.0.6167.85"
 pkgrel = 0
 archs = ["aarch64", "ppc64le", "x86_64"]
 configure_args = [
     'custom_toolchain="//build/toolchain/linux/unbundle:default"',
     'host_toolchain="//build/toolchain/linux/unbundle:default"',
-    'host_pkg_config="/usr/bin/pkg-config"',
+    "blink_enable_generated_code_formatting=false",
     "blink_symbol_level=0",
-    "symbol_level=0",
     "chrome_pgo_phase=0",
     'clang_base_path="/usr"',
     "clang_use_chrome_plugins=false",
-    'rust_sysroot_absolute="/usr"',
-    "treat_warnings_as_errors=false",
-    "fatal_linker_warnings=false",
+    # defaults to whatever latest version they have and is used for some conf paths
+    'clang_version="17"',
```

Comment:
we could set this from `clang --version` output since it's just the majorver that is used to point to the clang-rt libpath. i did this because it was easier and was too lazy to do some regex

Reply:
how does chromium use this?

Reply:
toolchain.gni has it hardcoded:  

```
declare_args() {
  if (llvm_android_mainline) {  # https://crbug.com/1481060
    clang_version = "17"
  } else {
    clang_version = "18"
  }
}
```

(they use master and bump this with bumps),

then for usage in `/build/config/clang/BUILD.gn`:

```
template("clang_lib") {
[...]
      _clang_lib_dir = "$clang_base_path/lib/clang/$clang_version/lib"
      _lib_file = "${_prefix}clang_rt.${_libname}${_suffix}.${_ext}"
      libs = [ "$_clang_lib_dir/$_dir/$_lib_file" ]
```

`_suffix` is the target we patch to our name

clang_lib is used for

```
# Adds a dependency on the Clang runtime library clang_rt.builtins.
clang_lib("compiler_builtins") {

[..]

# Adds a dependency on the Clang runtime library clang_rt.profile.
clang_lib("compiler_profile") {
```

etc. adding the rt stuff for linking for sanitizer/profile related stuff, you've probably gone over it before

Reply:
but yeah it's just the number in `usr/lib/clang/17/`

Reply:
i don't think chromium should by itself be linking any sanitizers though, at least not in the distro configuration

Reply:
and the builtins library should also be irrelevant to chromium, that's just implicit to clang

Reply:
shrug. i don't think any actual sanitizers are in use via manually linking the .a with the current set of args (enabling cfi does use some of these iirc)

somewhere in the ninja dep graph a dep on those paths is added so if they're wrong you get the usual /usr/some/wrong/path not found from ninja (replace 17 with 18 or remove the _suffix updates etc and see). one example target in question:  

```
build obj/third_party/webrtc/api/video_codecs/librtc_software_fallback_wrappers.a: alink obj/third_party/webrtc/api/video_codecs/rtc_software_fallback_wrappers/video_decoder_software_fallback_wrapper.o obj/third_party/webrtc/api/video_codecs/rtc_software_fallback_wrappers/video_encoder_software_fallback_wrapper.o | ../../../../usr/lib/clang/17/lib/x86_64-chimera-linux-musl/libclang_rt.builtins.a || obj/base/libbase.a obj/third_party/webrtc/api/video_codecs/libvideo_codecs_api.a obj/third_party/webrtc/api/fec_controller_api.stamp obj/third_party/webrtc/api/transport/libfield_trial_based_config.a obj/third_party/webrtc/api/video/libvideo_frame.a obj/third_party/webrtc/media/librtc_media_base.a obj/third_party/webrtc/modules/video_coding/libvideo_codec_interface.a obj/third_party/webrtc/modules/video_coding/libvideo_coding_utility.a obj/third_party/webrtc/rtc_base/libchecks.a obj/third_party/webrtc/rtc_base/libevent_tracer.a obj/third_party/webrtc/rtc_base/liblogging.a obj/third_party/webrtc/rtc_base/experiments/libfield_trial_parser.a obj/third_party/webrtc/rtc_base/system/rtc_export.stamp obj/third_party/webrtc/system_wrappers/libfield_trial.a obj/third_party/webrtc/system_wrappers/libmetrics.a obj/third_party/webrtc/api/video/libencoded_image.a obj/third_party/webrtc/api/video/libvideo_bitrate_allocation.a obj/third_party/webrtc/api/video/libvideo_rtp_headers.a obj/third_party/abseil-cpp/absl.stamp obj/base/libbase_static.a obj/base/allocator/partition_allocator/src/partition_alloc/libraw_ptr.a obj/base/allocator/partition_allocator/src/partition_alloc/liballocator_base.a obj/base/allocator/partition_allocator/src/partition_alloc/liballocator_core.a obj/base/allocator/partition_allocator/src/partition_alloc/liballocator_shim.a obj/base/numerics/base_numerics.stamp obj/third_party/abseil-cpp/absl/algorithm/container.stamp obj/third_party/abseil-cpp/absl/algorithm/algorithm.stamp obj/third_party/abseil-cpp/absl/base/config.stamp obj/third_party/abseil-cpp/absl/base/core_headers.stamp obj/third_party/abseil-cpp/absl/meta/type_traits.stamp obj/third_party/abseil-cpp/absl/base/base.stamp obj/third_party/abseil-cpp/absl/base/atomic_hook.stamp obj/third_party/abseil-cpp/absl/base/base_internal.stamp obj/third_party/abseil-cpp/absl/base/cycleclock_internal.stamp obj/third_party/abseil-cpp/absl/base/dynamic_annotations.stamp obj/third_party/abseil-cpp/absl/base/log_severity.stamp obj/third_party/abseil-cpp/absl/base/raw_logging_internal.stamp obj/third_party/abseil-cpp/absl/base/errno_saver.stamp obj/third_party/abseil-cpp/absl/base/spinlock_wait.stamp obj/third_party/abseil-cpp/absl/base/no_destructor.stamp obj/third_party/abseil-cpp/absl/base/nullability.stamp obj/third_party/abseil-cpp/absl/base/prefetch.stamp obj/third_party/abseil-cpp/absl/cleanup/cleanup.stamp obj/third_party/abseil-cpp/absl/cleanup/cleanup_internal.stamp obj/third_party/abseil-cpp/absl/utility/utility.stamp obj/third_party/abseil-cpp/absl/container/btree.stamp obj/third_party/abseil-cpp/absl/container/common.stamp obj/third_party/abseil-cpp/absl/types/optional.stamp obj/third_party/abseil-cpp/absl/types/bad_optional_access.stamp obj/third_party/abseil-cpp/absl/memory/memory.stamp obj/third_party/abseil-cpp/absl/container/common_policy_traits.stamp obj/third_party/abseil-cpp/absl/container/compressed_tuple.stamp obj/third_party/abseil-cpp/absl/container/container_memory.stamp obj/third_party/abseil-cpp/absl/container/layout.stamp obj/third_party/abseil-cpp/absl/debugging/demangle_internal.stamp obj/third_party/abseil-cpp/absl/strings/strings.stamp obj/third_party/abseil-cpp/absl/strings/string_view.stamp obj/third_party/abseil-cpp/absl/base/throw_delegate.stamp obj/third_party/abseil-cpp/absl/strings/charset.stamp obj/third_party/abseil-cpp/absl/strings/internal.stamp obj/third_party/abseil-cpp/absl/base/endian.stamp obj/third_party/abseil-cpp/absl/numeric/bits.stamp obj/third_party/abseil-cpp/absl/numeric/int128.stamp obj/third_party/abseil-cpp/absl/types/span.stamp obj/third_party/abseil-cpp/absl/strings/cord.stamp obj/third_party/abseil-cpp/absl/strings/cord_internal.stamp obj/third_party/abseil-cpp/absl/container/inlined_vector.stamp obj/third_party/abseil-cpp/absl/container/inlined_vector_internal.stamp obj/third_party/abseil-cpp/absl/crc/crc_cord_state.stamp obj/third_party/abseil-cpp/absl/crc/crc32c.stamp obj/third_party/abseil-cpp/absl/crc/cpu_detect.stamp obj/third_party/abseil-cpp/absl/crc/crc_internal.stamp obj/third_party/abseil-cpp/absl/crc/non_temporal_memcpy.stamp obj/third_party/abseil-cpp/absl/crc/non_temporal_arm_intrinsics.stamp obj/third_party/abseil-cpp/absl/strings/str_format.stamp obj/third_party/abseil-cpp/absl/strings/str_format_internal.stamp obj/third_party/abseil-cpp/absl/functional/function_ref.stamp obj/third_party/abseil-cpp/absl/functional/any_invocable.stamp obj/third_party/abseil-cpp/absl/numeric/representation.stamp obj/third_party/abseil-cpp/absl/strings/cordz_functions.stamp obj/third_party/abseil-cpp/absl/profiling/exponential_biased.stamp obj/third_party/abseil-cpp/absl/strings/cordz_info.stamp obj/third_party/abseil-cpp/absl/strings/cordz_handle.stamp obj/third_party/abseil-cpp/absl/synchronization/synchronization.stamp obj/third_party/abseil-cpp/absl/synchronization/graphcycles_internal.stamp obj/third_party/abseil-cpp/absl/base/malloc_internal.stamp obj/third_party/abseil-cpp/absl/synchronization/kernel_timeout_internal.stamp obj/third_party/abseil-cpp/absl/time/time.stamp obj/third_party/abseil-cpp/absl/time/internal/cctz/civil_time.stamp obj/third_party/abseil-cpp/absl/time/internal/cctz/time_zone.stamp obj/third_party/abseil-cpp/absl/debugging/stacktrace.stamp obj/third_party/abseil-cpp/absl/debugging/debugging_internal.stamp obj/third_party/abseil-cpp/absl/debugging/symbolize.stamp obj/third_party/abseil-cpp/absl/strings/cordz_statistics.stamp obj/third_party/abseil-cpp/absl/strings/cordz_update_tracker.stamp obj/third_party/abseil-cpp/absl/strings/cordz_update_scope.stamp obj/third_party/abseil-cpp/absl/types/compare.stamp obj/third_party/abseil-cpp/absl/container/fixed_array.stamp obj/third_party/abseil-cpp/absl/container/flat_hash_map.stamp obj/third_party/abseil-cpp/absl/container/hash_function_defaults.stamp obj/third_party/abseil-cpp/absl/hash/hash.stamp obj/third_party/abseil-cpp/absl/hash/city.stamp obj/third_party/abseil-cpp/absl/hash/low_level_hash.stamp obj/third_party/abseil-cpp/absl/types/variant.stamp obj/third_party/abseil-cpp/absl/types/bad_variant_access.stamp obj/third_party/abseil-cpp/absl/container/raw_hash_map.stamp obj/third_party/abseil-cpp/absl/container/raw_hash_set.stamp obj/third_party/abseil-cpp/absl/container/hash_policy_traits.stamp obj/third_party/abseil-cpp/absl/container/hashtable_debug_hooks.stamp obj/third_party/abseil-cpp/absl/container/hashtablez_sampler.stamp obj/third_party/abseil-cpp/absl/profiling/sample_recorder.stamp obj/third_party/abseil-cpp/absl/container/flat_hash_set.stamp obj/third_party/abseil-cpp/absl/container/node_hash_map.stamp obj/third_party/abseil-cpp/absl/container/node_slot_policy.stamp obj/third_party/abseil-cpp/absl/container/node_hash_set.stamp obj/third_party/abseil-cpp/absl/debugging/failure_signal_handler.stamp obj/third_party/abseil-cpp/absl/debugging/examine_stack.stamp obj/third_party/abseil-cpp/absl/functional/bind_front.stamp obj/third_party/abseil-cpp/absl/functional/overload.stamp obj/third_party/abseil-cpp/absl/log/absl_check.stamp obj/third_party/abseil-cpp/absl/log/internal/check_impl.stamp obj/third_party/abseil-cpp/absl/log/internal/check_op.stamp obj/third_party/abseil-cpp/absl/log/internal/nullguard.stamp obj/third_party/abseil-cpp/absl/log/internal/nullstream.stamp obj/third_party/abseil-cpp/absl/log/internal/strip.stamp obj/third_party/abseil-cpp/absl/log/internal/log_message.stamp obj/third_party/abseil-cpp/absl/log/internal/append_truncated.stamp obj/third_party/abseil-cpp/absl/log/internal/format.stamp obj/third_party/abseil-cpp/absl/log/internal/config.stamp obj/third_party/abseil-cpp/absl/log/internal/globals.stamp obj/third_party/abseil-cpp/absl/log/internal/log_sink_set.stamp obj/third_party/abseil-cpp/absl/log/globals.stamp obj/third_party/abseil-cpp/absl/log/log_entry.stamp obj/third_party/abseil-cpp/absl/log/log_sink.stamp obj/third_party/abseil-cpp/absl/log/internal/proto.stamp obj/third_party/abseil-cpp/absl/base/strerror.stamp obj/third_party/abseil-cpp/absl/log/log_sink_registry.stamp obj/third_party/abseil-cpp/absl/log/internal/conditions.stamp obj/third_party/abseil-cpp/absl/log/internal/voidify.stamp obj/third_party/abseil-cpp/absl/log/absl_log.stamp obj/third_party/abseil-cpp/absl/log/internal/log_impl.stamp obj/third_party/abseil-cpp/absl/log/die_if_null.stamp obj/third_party/abseil-cpp/absl/log/log.stamp obj/third_party/abseil-cpp/absl/random/random.stamp obj/third_party/abseil-cpp/absl/random/distributions.stamp obj/third_party/abseil-cpp/absl/random/internal/distribution_caller.stamp obj/third_party/abseil-cpp/absl/base/fast_type_id.stamp obj/third_party/abseil-cpp/absl/random/internal/fast_uniform_bits.stamp obj/third_party/abseil-cpp/absl/random/internal/traits.stamp obj/third_party/abseil-cpp/absl/random/internal/fastmath.stamp obj/third_party/abseil-cpp/absl/random/internal/generate_real.stamp obj/third_party/abseil-cpp/absl/random/internal/iostream_state_saver.stamp obj/third_party/abseil-cpp/absl/random/internal/uniform_helper.stamp obj/third_party/abseil-cpp/absl/random/internal/wide_multiply.stamp obj/third_party/abseil-cpp/absl/random/seed_sequences.stamp obj/third_party/abseil-cpp/absl/random/seed_gen_exception.stamp obj/third_party/abseil-cpp/absl/random/internal/pool_urbg.stamp obj/third_party/abseil-cpp/absl/random/internal/randen.stamp obj/third_party/abseil-cpp/absl/random/internal/platform.stamp obj/third_party/abseil-cpp/absl/random/internal/randen_hwaes.stamp obj/third_party/abseil-cpp/absl/random/internal/randen_hwaes_impl.stamp obj/third_party/abseil-cpp/absl/random/internal/randen_slow.stamp obj/third_party/abseil-cpp/absl/random/internal/seed_material.stamp obj/third_party/abseil-cpp/absl/random/internal/salted_seed_seq.stamp obj/third_party/abseil-cpp/absl/random/internal/nonsecure_base.stamp obj/third_party/abseil-cpp/absl/random/internal/pcg_engine.stamp obj/third_party/abseil-cpp/absl/random/internal/randen_engine.stamp obj/third_party/abseil-cpp/absl/status/status.stamp obj/third_party/abseil-cpp/absl/status/statusor.stamp obj/third_party/abseil-cpp/absl/strings/has_ostream_operator.stamp obj/build/rust/libcxx_cppdeps.a obj/third_party/rust/cxx/v1/lib/libcxx-cxx-1.rlib obj/build/rust/std/remap_alloc.stamp obj/third_party/boringssl/libboringssl.a obj/third_party/boringssl/boringssl_asm.stamp obj/third_party/boringssl/src/third_party/fiat/fiat_license.stamp obj/base/tracing/protos/libchrome_track_event_zero.a obj/third_party/perfetto/libperfetto.stamp obj/third_party/perfetto/include/perfetto/ext/trace_processor/importers/memory_tracker/memory_tracker.stamp obj/third_party/perfetto/include/perfetto/base/base.stamp obj/third_party/perfetto/include/perfetto/public/base.stamp obj/third_party/perfetto/include/perfetto/public/abi/base.stamp obj/third_party/perfetto/include/perfetto/ext/base/base.stamp obj/third_party/perfetto/include/perfetto/ext/tracing/core/core.stamp obj/third_party/perfetto/protos/perfetto/common/libcpp.a obj/third_party/perfetto/src/protozero/protozero.stamp obj/third_party/perfetto/include/perfetto/protozero/protozero.stamp obj/third_party/perfetto/include/perfetto/public/protozero.stamp obj/third_party/perfetto/src/base/base.stamp obj/third_party/perfetto/src/base/perfetto_base_default_platform.stamp obj/third_party/perfetto/include/perfetto/tracing/core/core.stamp obj/third_party/perfetto/include/perfetto/tracing/core/forward_decls.stamp obj/third_party/perfetto/protos/perfetto/config/libcpp.a obj/third_party/perfetto/protos/perfetto/config/android/libcpp.a obj/third_party/perfetto/protos/perfetto/config/etw/libcpp.a obj/third_party/perfetto/protos/perfetto/config/ftrace/libcpp.a obj/third_party/perfetto/protos/perfetto/config/gpu/libcpp.a obj/third_party/perfetto/protos/perfetto/config/inode_file/libcpp.a obj/third_party/perfetto/protos/perfetto/config/interceptors/libcpp.a obj/third_party/perfetto/protos/perfetto/config/power/libcpp.a obj/third_party/perfetto/protos/perfetto/config/process_stats/libcpp.a obj/third_party/perfetto/protos/perfetto/config/profiling/libcpp.a obj/third_party/perfetto/protos/perfetto/config/statsd/libcpp.a obj/third_party/perfetto/protos/perfetto/config/sys_stats/libcpp.a obj/third_party/perfetto/protos/perfetto/config/system_info/libcpp.a obj/third_party/perfetto/protos/perfetto/config/track_event/libcpp.a obj/third_party/perfetto/include/perfetto/tracing/tracing.stamp obj/third_party/perfetto/protos/perfetto/common/libzero.a obj/third_party/perfetto/protos/perfetto/trace/libminimal_zero.a obj/third_party/perfetto/protos/perfetto/config/libzero.a obj/third_party/perfetto/protos/perfetto/config/android/libzero.a obj/third_party/perfetto/protos/perfetto/config/etw/libzero.a obj/third_party/perfetto/protos/perfetto/config/ftrace/libzero.a obj/third_party/perfetto/protos/perfetto/config/gpu/libzero.a obj/third_party/perfetto/protos/perfetto/config/inode_file/libzero.a obj/third_party/perfetto/protos/perfetto/config/interceptors/libzero.a obj/third_party/perfetto/protos/perfetto/config/power/libzero.a obj/third_party/perfetto/protos/perfetto/config/process_stats/libzero.a obj/third_party/perfetto/protos/perfetto/config/profiling/libzero.a obj/third_party/perfetto/protos/perfetto/config/statsd/libzero.a obj/third_party/perfetto/protos/perfetto/config/sys_stats/libzero.a obj/third_party/perfetto/protos/perfetto/config/system_info/libzero.a obj/third_party/perfetto/protos/perfetto/config/track_event/libzero.a obj/third_party/perfetto/protos/perfetto/trace/libnon_minimal_zero.a obj/third_party/perfetto/protos/perfetto/trace/android/libzero.a obj/third_party/perfetto/protos/perfetto/trace/chrome/libzero.a obj/third_party/perfetto/protos/perfetto/trace/etw/libzero.a obj/third_party/perfetto/protos/perfetto/trace/filesystem/libzero.a obj/third_party/perfetto/protos/perfetto/trace/ftrace/libzero.a obj/third_party/perfetto/protos/perfetto/trace/gpu/libzero.a obj/third_party/perfetto/protos/perfetto/trace/interned_data/libzero.a obj/third_party/perfetto/protos/perfetto/trace/profiling/libzero.a obj/third_party/perfetto/protos/perfetto/trace/track_event/libzero.a obj/third_party/perfetto/protos/perfetto/trace/perfetto/libzero.a obj/third_party/perfetto/protos/perfetto/trace/power/libzero.a obj/third_party/perfetto/protos/perfetto/trace/ps/libzero.a obj/third_party/perfetto/protos/perfetto/trace/statsd/libzero.a obj/third_party/perfetto/protos/perfetto/trace/sys_stats/libzero.a obj/third_party/perfetto/protos/perfetto/trace/system_info/libzero.a obj/third_party/perfetto/protos/perfetto/trace/translation/libzero.a obj/third_party/perfetto/protos/perfetto/trace/track_event/libcpp.a obj/third_party/perfetto/include/perfetto/ext/tracing/ipc/ipc.stamp obj/third_party/perfetto/include/perfetto/ext/trace_processor/export_json.stamp obj/third_party/perfetto/include/perfetto/trace_processor/storage.stamp obj/third_party/perfetto/include/perfetto/trace_processor/basic_types.stamp obj/third_party/perfetto/src/trace_processor/importers/memory_tracker/graph_processor.stamp obj/third_party/perfetto/src/tracing/client_api_without_backends.stamp obj/third_party/perfetto/src/tracing/core/core.stamp obj/third_party/perfetto/src/tracing/common.stamp obj/third_party/perfetto/src/tracing/in_process_backend.stamp obj/third_party/perfetto/src/tracing/core/service.stamp obj/third_party/perfetto/src/android_stats/android_stats.stamp obj/third_party/perfetto/src/android_stats/perfetto_atoms.stamp obj/third_party/perfetto/src/base/version.stamp obj/third_party/perfetto/include/perfetto/ext/base/version.stamp obj/third_party/perfetto/src/protozero/filtering/message_filter.stamp obj/third_party/perfetto/src/protozero/filtering/bytecode_parser.stamp obj/third_party/perfetto/src/protozero/filtering/bytecode_common.stamp obj/third_party/perfetto/src/protozero/filtering/string_filter.stamp obj/third_party/perfetto/src/tracing/system_backend.stamp obj/third_party/perfetto/src/tracing/ipc/common.stamp obj/third_party/perfetto/src/tracing/ipc/default_socket.stamp obj/third_party/perfetto/include/perfetto/ext/ipc/ipc.stamp obj/third_party/perfetto/src/tracing/ipc/producer/producer.stamp obj/third_party/perfetto/protos/perfetto/ipc/libipc.a obj/third_party/perfetto/src/ipc/common.stamp obj/third_party/perfetto/protos/perfetto/ipc/libwire_protocol_cpp.a obj/third_party/perfetto/protos/perfetto/ipc/libcpp.a obj/third_party/perfetto/src/ipc/client.stamp obj/third_party/perfetto/src/base/unix_socket.stamp obj/third_party/perfetto/src/tracing/ipc/service/service.stamp obj/third_party/perfetto/src/ipc/host.stamp obj/third_party/perfetto/src/tracing/ipc/consumer/consumer.stamp obj/third_party/perfetto/src/tracing/platform_impl.stamp obj/third_party/perfetto/src/trace_processor/export_json.stamp obj/third_party/perfetto/src/trace_processor/storage_minimal.stamp obj/third_party/perfetto/src/trace_processor/containers/containers.stamp obj/third_party/perfetto/protos/perfetto/trace_processor/libzero.a obj/third_party/perfetto/src/trace_processor/importers/common/common.stamp obj/third_party/perfetto/src/trace_processor/importers/common/trace_parser_hdr.stamp obj/third_party/perfetto/src/trace_processor/util/proto_to_args_parser.stamp obj/third_party/perfetto/src/trace_processor/util/descriptors.stamp obj/third_party/perfetto/src/trace_processor/util/util.stamp obj/third_party/perfetto/src/trace_processor/util/interned_message_view.stamp obj/third_party/perfetto/include/perfetto/trace_processor/trace_processor.stamp obj/third_party/perfetto/src/trace_processor/util/protozero_to_text.stamp obj/third_party/perfetto/src/trace_processor/importers/common/parser_types.stamp obj/third_party/perfetto/src/trace_processor/importers/proto/packet_sequence_state_generation_hdr.stamp obj/third_party/perfetto/src/trace_processor/storage/storage.stamp obj/third_party/perfetto/src/trace_processor/tables/tables.stamp obj/third_party/perfetto/src/trace_processor/db/db.stamp obj/third_party/perfetto/src/trace_processor/util/glob.stamp obj/third_party/perfetto/src/trace_processor/util/regex.stamp obj/third_party/perfetto/src/trace_processor/db/overlays/overlays.stamp obj/third_party/perfetto/src/trace_processor/db/storage/storage.stamp obj/third_party/perfetto/src/trace_processor/metatrace.stamp obj/third_party/perfetto/src/trace_processor/types/types.stamp obj/third_party/perfetto/src/trace_processor/views/views.stamp obj/third_party/perfetto/src/trace_processor/importers/fuchsia/fuchsia_record.stamp obj/third_party/perfetto/src/trace_processor/importers/systrace/systrace_line.stamp obj/third_party/perfetto/src/trace_processor/importers/ftrace/minimal.stamp obj/third_party/perfetto/src/trace_processor/importers/proto/proto_importer_module.stamp obj/third_party/perfetto/src/trace_processor/importers/proto/minimal.stamp obj/third_party/perfetto/src/trace_processor/sorter/sorter.stamp obj/third_party/perfetto/src/trace_processor/util/bump_allocator.stamp obj/third_party/perfetto/src/trace_processor/util/gzip.stamp obj/third_party/zlib/zlib.stamp obj/third_party/perfetto/src/trace_processor/util/stack_traces_util.stamp obj/third_party/perfetto/src/trace_processor/importers/json/minimal.stamp obj/third_party/jsoncpp/jsoncpp.stamp obj/base/third_party/double_conversion/double_conversion.stamp obj/base/third_party/dynamic_annotations/libdynamic_annotations.a obj/third_party/modp_b64/libmodp_b64.a obj/base/third_party/symbolize/libsymbolize.a obj/base/third_party/xdg_user_dirs/libxdg_user_dirs.a obj/third_party/libevent/libevent.stamp obj/third_party/icu/icui18n.stamp obj/third_party/icu/icuuc.stamp obj/third_party/webrtc/api/video_codecs/scalability_mode.stamp obj/third_party/webrtc/rtc_base/safe_compare.stamp obj/third_party/webrtc/rtc_base/type_traits.stamp obj/third_party/webrtc/api/scoped_refptr.stamp obj/third_party/webrtc/rtc_base/system/inline.stamp obj/third_party/webrtc/modules/module_fec_api.stamp obj/third_party/webrtc/api/video/video_frame_type.stamp obj/third_party/webrtc/api/array_view.stamp obj/third_party/webrtc/modules/video_coding/codec_globals_headers.stamp obj/third_party/webrtc/rtc_base/macromagic.stamp obj/third_party/webrtc/rtc_base/system/arch.stamp obj/third_party/webrtc/rtc_base/libplatform_thread_types.a obj/third_party/webrtc/rtc_base/libstringutils.a obj/third_party/webrtc/rtc_base/safe_minmax.stamp obj/third_party/webrtc/rtc_base/libtimeutils.a obj/third_party/webrtc/rtc_base/safe_conversions.stamp obj/third_party/webrtc/api/units/libtimestamp.a obj/third_party/webrtc/api/units/libtime_delta.a obj/third_party/webrtc/rtc_base/units/unit_base.stamp obj/third_party/webrtc/rtc_base/divide_round.stamp obj/third_party/webrtc/rtc_base/synchronization/mutex.stamp obj/third_party/webrtc/rtc_base/synchronization/libyield.a obj/third_party/webrtc/rtc_base/system/no_unique_address.stamp obj/third_party/webrtc/rtc_base/refcount.stamp obj/third_party/webrtc/api/ref_count.stamp obj/third_party/webrtc/api/units/libdata_rate.a obj/third_party/webrtc/api/units/libdata_size.a obj/third_party/webrtc/api/units/libfrequency.a obj/third_party/webrtc/api/video/video_codec_constants.stamp obj/third_party/webrtc/api/make_ref_counted.stamp obj/third_party/webrtc/api/librtp_packet_info.a obj/third_party/webrtc/api/refcountedbase.stamp obj/third_party/webrtc/api/librtp_headers.a obj/third_party/webrtc/api/video_track_source_constraints.stamp obj/third_party/webrtc/rtc_base/memory/libaligned_malloc.a obj/third_party/libyuv/libyuv_internal.a obj/third_party/webrtc/api/video/render_resolution.stamp obj/third_party/webrtc/api/video/resolution.stamp obj/third_party/webrtc/api/field_trials_registry.stamp obj/third_party/webrtc/api/field_trials_view.stamp obj/third_party/webrtc/experiments/registered_field_trials.stamp obj/third_party/webrtc/rtc_base/containers/flat_set.stamp obj/third_party/webrtc/rtc_base/containers/libflat_containers_internal.a obj/third_party/webrtc/media/libcodec.a obj/third_party/webrtc/media/libmedia_constants.a obj/third_party/webrtc/api/librtp_parameters.a obj/third_party/webrtc/api/priority.stamp obj/third_party/webrtc/api/rtp_transceiver_direction.stamp obj/third_party/webrtc/api/audio_codecs/libaudio_codecs_api.a obj/third_party/webrtc/api/bitrate_allocation.stamp obj/third_party/webrtc/rtc_base/buffer.stamp obj/third_party/webrtc/rtc_base/libzero_memory.a obj/third_party/webrtc/rtc_base/libplatform_thread.a obj/third_party/webrtc/rtc_base/librtc_event.a obj/third_party/webrtc/api/sequence_checker.stamp obj/third_party/webrtc/rtc_base/synchronization/libsequence_checker_internal.a obj/third_party/webrtc/api/task_queue/libtask_queue.a obj/third_party/webrtc/api/location.stamp obj/third_party/webrtc/rtc_base/sanitizer.stamp obj/third_party/webrtc/media/media_channel.stamp obj/third_party/webrtc/media/librtp_utils.a obj/third_party/webrtc/media/libturn_utils.a obj/third_party/webrtc/api/transport/stun_types.stamp obj/third_party/webrtc/rtc_base/libbyte_buffer.a obj/third_party/webrtc/rtc_base/byte_order.stamp obj/third_party/webrtc/rtc_base/libcrc32.a obj/third_party/webrtc/rtc_base/libip_address.a obj/third_party/webrtc/rtc_base/libnet_helpers.a obj/third_party/webrtc/rtc_base/libsocket_address.a obj/third_party/webrtc/rtc_base/libssl.a obj/third_party/webrtc/rtc_base/libasync_socket.a obj/third_party/webrtc/rtc_base/libsocket.a obj/third_party/webrtc/rtc_base/third_party/sigslot/libsigslot.a obj/third_party/webrtc/rtc_base/libcopy_on_write_buffer.a obj/third_party/webrtc/rtc_base/libstream.a obj/third_party/webrtc/rtc_base/libthreading.a obj/third_party/webrtc/rtc_base/libasync_dns_resolver.a obj/third_party/webrtc/api/async_dns_resolver.stamp obj/third_party/webrtc/api/task_queue/libpending_task_safety_flag.a obj/third_party/webrtc/rtc_base/libcriticalsection.a obj/third_party/webrtc/rtc_base/system/unused.stamp obj/third_party/webrtc/rtc_base/network_constants.stamp obj/third_party/webrtc/rtc_base/libnull_socket_server.a obj/third_party/webrtc/rtc_base/socket_server.stamp obj/third_party/webrtc/rtc_base/socket_factory.stamp obj/third_party/webrtc/rtc_base/librtc_task_queue.a obj/third_party/webrtc/api/function_view.stamp obj/third_party/webrtc/rtc_base/task_utils/librepeating_task.a obj/third_party/webrtc/system_wrappers/libsystem_wrappers.a obj/third_party/webrtc/modules/module_api_public.stamp obj/third_party/webrtc/rtc_base/librtc_numerics.a obj/third_party/webrtc/rtc_base/mod_ops.stamp obj/third_party/webrtc/rtc_base/third_party/base64/libbase64.a obj/third_party/webrtc/modules/rtp_rtcp/librtp_rtcp_format.a obj/third_party/webrtc/modules/rtp_rtcp/libleb128.a obj/third_party/webrtc/api/transport/libnetwork_control.a obj/third_party/webrtc/api/rtc_event_log/librtc_event_log.a obj/third_party/webrtc/api/libjingle_logging_api.stamp obj/third_party/webrtc/api/environment/environment.stamp obj/third_party/webrtc/api/transport/rtp/dependency_descriptor.stamp obj/third_party/webrtc/api/video/video_layers_allocation.stamp obj/third_party/webrtc/rtc_base/libbit_buffer.a obj/third_party/webrtc/rtc_base/bitstream_reader.stamp obj/third_party/webrtc/rtc_base/libasync_packet_socket.a obj/third_party/webrtc/rtc_base/callback_list.stamp obj/third_party/webrtc/rtc_base/untyped_function.stamp obj/third_party/webrtc/rtc_base/system/assume.stamp obj/third_party/webrtc/rtc_base/dscp.stamp obj/third_party/webrtc/rtc_base/network/libreceived_packet.a obj/third_party/webrtc/rtc_base/network/libsent_packet.a obj/third_party/webrtc/media/libstream_params.a obj/third_party/webrtc/media/librid_description.a obj/third_party/webrtc/rtc_base/libunique_id_generator.a obj/third_party/webrtc/api/libaudio_options_api.a obj/third_party/webrtc/api/call_api.stamp obj/third_party/webrtc/api/frame_transformer_interface.stamp obj/third_party/webrtc/api/video/libencoded_frame.a obj/third_party/webrtc/modules/rtp_rtcp/librtp_video_header.a obj/third_party/webrtc/api/video/video_frame_metadata.stamp obj/third_party/webrtc/common_video/libcommon_video.a obj/third_party/webrtc/api/video/libvideo_bitrate_allocator.a obj/third_party/webrtc/api/video/libvideo_frame_i010.a obj/third_party/webrtc/api/video_codecs/bitstream_parser_api.stamp obj/third_party/webrtc/rtc_base/librace_checker.a obj/third_party/webrtc/rtc_base/librate_statistics.a obj/third_party/webrtc/rtc_base/compile_assert_c.stamp obj/third_party/webrtc/rtc_base/containers/flat_map.stamp obj/third_party/webrtc/common_video/generic_frame_descriptor/libgeneric_frame_descriptor.a obj/third_party/webrtc/api/libmedia_stream_interface.a obj/third_party/webrtc/modules/audio_processing/libaudio_processing_statistics.a obj/third_party/webrtc/api/video/recordable_encoded_frame.stamp obj/third_party/webrtc/api/librtc_error.a obj/third_party/webrtc/api/librtp_sender_interface.a obj/third_party/webrtc/api/libdtls_transport_interface.a obj/third_party/webrtc/api/ice_transport_interface.stamp obj/third_party/webrtc/api/packet_socket_factory.stamp obj/third_party/webrtc/rtc_base/libproxy_info.a obj/third_party/webrtc/rtc_base/libcrypt_string.a obj/third_party/webrtc/api/dtmf_sender_interface.stamp obj/third_party/webrtc/api/crypto/frame_encryptor_interface.stamp obj/third_party/webrtc/api/crypto/frame_decryptor_interface.stamp obj/third_party/webrtc/api/transport/datagram_transport_interface.stamp obj/third_party/webrtc/api/transport/rtp/rtp_source.stamp obj/third_party/webrtc/call/libvideo_stream_api.a obj/third_party/webrtc/call/receive_stream_interface.stamp obj/third_party/webrtc/call/librtp_interfaces.a obj/third_party/webrtc/api/network_state_predictor_api.stamp obj/third_party/webrtc/api/crypto/liboptions.a obj/third_party/webrtc/api/transport/libbitrate_settings.a obj/third_party/webrtc/common_video/frame_counts.stamp obj/third_party/webrtc/api/libtransport_api.a obj/third_party/webrtc/api/adaptation/resource_adaptation_api.stamp obj/third_party/webrtc/api/video/video_stream_encoder.stamp obj/third_party/webrtc/api/video/libvideo_adaptation.a obj/third_party/webrtc/api/video/video_bitrate_allocator_factory.stamp obj/third_party/webrtc/video/config/libencoder_config.a obj/third_party/webrtc/rtc_base/libnetwork_route.a obj/third_party/webrtc/media/libmedia_channel_impl.a obj/third_party/webrtc/media/rtc_media_config.stamp obj/third_party/webrtc/api/audio/audio_frame_processor.stamp obj/third_party/webrtc/call/libcall_interfaces.a obj/third_party/webrtc/call/audio_sender_interface.stamp obj/third_party/webrtc/api/audio/libaudio_frame_api.a obj/third_party/webrtc/api/audio/audio_mixer_api.stamp obj/third_party/webrtc/api/metronome/metronome.stamp obj/third_party/webrtc/api/neteq/neteq_api.stamp obj/third_party/webrtc/modules/async_audio_processing/libasync_audio_processing.a obj/third_party/webrtc/modules/audio_device/audio_device.stamp obj/third_party/webrtc/modules/audio_device/audio_device_api.stamp obj/third_party/webrtc/modules/audio_device/audio_device_impl.stamp obj/third_party/webrtc/modules/audio_device/libaudio_device_buffer.a obj/third_party/webrtc/common_audio/libcommon_audio_c.a obj/third_party/webrtc/common_audio/common_audio_c_arm_asm.stamp obj/third_party/webrtc/common_audio/libcommon_audio_cc.a obj/third_party/webrtc/common_audio/third_party/ooura/libfft_size_256.a obj/third_party/webrtc/common_audio/third_party/spl_sqrt_floor/libspl_sqrt_floor.a obj/third_party/webrtc/rtc_base/libtimestamp_aligner.a obj/third_party/webrtc/modules/audio_device/audio_device_config.stamp obj/third_party/webrtc/modules/audio_device/audio_device_default.stamp obj/third_party/webrtc/modules/audio_device/libaudio_device_dummy.a obj/third_party/webrtc/modules/audio_device/libaudio_device_generic.a obj/third_party/webrtc/common_audio/libcommon_audio.a obj/third_party/webrtc/common_audio/sinc_resampler.stamp obj/third_party/webrtc/rtc_base/gtest_prod.stamp obj/third_party/webrtc/rtc_base/system/libfile_wrapper.a obj/third_party/webrtc/common_audio/libcommon_audio_sse2.a obj/third_party/webrtc/common_audio/fir_filter.stamp obj/third_party/webrtc/common_audio/libcommon_audio_avx2.a obj/third_party/webrtc/rtc_base/librandom.a obj/third_party/webrtc/modules/utility/utility.stamp obj/third_party/webrtc/modules/audio_processing/libaudio_processing.a obj/third_party/webrtc/modules/audio_processing/aec_dump_interface.stamp obj/third_party/webrtc/modules/audio_processing/libapi.a obj/third_party/webrtc/modules/audio_processing/audio_frame_view.stamp obj/third_party/webrtc/api/audio/libaec3_config.a obj/third_party/webrtc/api/audio/echo_control.stamp obj/third_party/webrtc/modules/audio_processing/agc/gain_control_interface.stamp obj/third_party/webrtc/modules/audio_processing/libapm_logging.a obj/third_party/webrtc/modules/audio_processing/libaudio_buffer.a obj/third_party/webrtc/modules/audio_processing/libaudio_frame_proxies.a obj/third_party/webrtc/modules/audio_processing/libgain_controller2.a obj/third_party/webrtc/modules/audio_processing/agc2/libadaptive_digital_gain_controller.a obj/third_party/webrtc/modules/audio_processing/agc2/common.stamp obj/third_party/webrtc/modules/audio_processing/agc2/libgain_applier.a obj/third_party/webrtc/modules/audio_processing/agc2/libcpu_features.a obj/third_party/webrtc/modules/audio_processing/agc2/libfixed_digital.a obj/third_party/webrtc/modules/audio_processing/agc2/libinput_volume_controller.a obj/third_party/webrtc/modules/audio_processing/agc2/libclipping_predictor.a obj/third_party/webrtc/modules/audio_processing/agc2/gain_map.stamp obj/third_party/webrtc/modules/audio_processing/agc2/libinput_volume_stats_reporter.a obj/third_party/webrtc/modules/audio_processing/agc2/libnoise_level_estimator.a obj/third_party/webrtc/modules/audio_processing/agc2/libbiquad_filter.a obj/third_party/webrtc/modules/audio_processing/agc2/libsaturation_protector.a obj/third_party/webrtc/modules/audio_processing/agc2/libspeech_level_estimator.a obj/third_party/webrtc/modules/audio_processing/agc2/libvad_wrapper.a obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/librnn_vad.a obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/rnn_vad_common.stamp obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/rnn_vad_layers.stamp obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/vector_math.stamp obj/third_party/rnnoise/rnn_vad.stamp obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/libvector_math_avx2.a obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/librnn_vad_lp_residual.a obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/librnn_vad_pitch.a obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/librnn_vad_auto_correlation.a obj/third_party/webrtc/modules/audio_processing/utility/libpffft_wrapper.a obj/third_party/pffft/libpffft.a obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/rnn_vad_sequence_buffer.stamp obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/librnn_vad_spectral_features.a obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/rnn_vad_ring_buffer.stamp obj/third_party/webrtc/modules/audio_processing/agc2/rnn_vad/rnn_vad_symmetric_matrix_buffer.stamp obj/third_party/webrtc/modules/audio_processing/libhigh_pass_filter.a obj/third_party/webrtc/modules/audio_processing/utility/libcascaded_biquad_filter.a obj/third_party/webrtc/modules/audio_processing/liboptionally_built_submodule_creators.a obj/third_party/webrtc/modules/audio_processing/transient/transient_suppressor_api.stamp obj/third_party/webrtc/modules/audio_processing/transient/libtransient_suppressor_impl.a obj/third_party/webrtc/modules/audio_processing/transient/libvoice_probability_delay_unit.a obj/third_party/webrtc/common_audio/libfir_filter_factory.a obj/third_party/webrtc/modules/audio_processing/rms_level.stamp obj/third_party/webrtc/audio/utility/libaudio_frame_operations.a obj/third_party/webrtc/rtc_base/swap_queue.stamp obj/third_party/webrtc/system_wrappers/libdenormal_disabler.a obj/third_party/webrtc/modules/audio_processing/aec3/libaec3.a obj/third_party/webrtc/modules/audio_processing/aec3/adaptive_fir_filter.stamp obj/third_party/webrtc/modules/audio_processing/aec3/aec3_common.stamp obj/third_party/webrtc/modules/audio_processing/aec3/aec3_fft.stamp obj/third_party/webrtc/modules/audio_processing/aec3/fft_data.stamp obj/third_party/webrtc/common_audio/third_party/ooura/libfft_size_128.a obj/third_party/webrtc/modules/audio_processing/aec3/render_buffer.stamp obj/third_party/webrtc/modules/audio_processing/aec3/adaptive_fir_filter_erl.stamp obj/third_party/webrtc/modules/audio_processing/aec3/matched_filter.stamp obj/third_party/webrtc/modules/audio_processing/aec3/vector_math.stamp obj/third_party/webrtc/modules/audio_processing/aec3/libaec3_avx2.a obj/third_party/webrtc/modules/audio_processing/aec_dump/aec_dump.stamp obj/third_party/webrtc/modules/audio_processing/aecm/libaecm_core.a obj/third_party/webrtc/modules/audio_processing/utility/liblegacy_delay_estimator.a obj/third_party/webrtc/modules/audio_processing/agc/libagc.a obj/third_party/webrtc/modules/audio_processing/agc/liblevel_estimation.a obj/third_party/webrtc/modules/audio_processing/vad/libvad.a obj/third_party/webrtc/modules/audio_coding/libisac_vad.a obj/third_party/webrtc/modules/audio_coding/isac_bwinfo.stamp obj/third_party/webrtc/rtc_base/system/ignore_warnings.stamp obj/third_party/webrtc/modules/third_party/fft/libfft.a obj/third_party/webrtc/modules/audio_processing/agc/liblegacy_agc.a obj/third_party/webrtc/modules/audio_processing/capture_levels_adjuster/libcapture_levels_adjuster.a obj/third_party/webrtc/modules/audio_processing/ns/libns.a obj/third_party/webrtc/modules/audio_processing/aec_dump/libaec_dump_impl.a obj/third_party/webrtc/rtc_base/protobuf_utils.stamp obj/third_party/protobuf/libprotobuf_lite.a obj/third_party/webrtc/modules/audio_processing/libaudioproc_debug_proto.a obj/third_party/webrtc/modules/rtp_rtcp/librtp_rtcp.a obj/third_party/webrtc/logging/librtc_event_rtp_rtcp.a obj/third_party/webrtc/logging/librtc_event_field.a obj/third_party/webrtc/logging/rtc_event_log_parse_status.stamp obj/third_party/webrtc/logging/librtc_event_number_encodings.a obj/third_party/webrtc/modules/audio_coding/audio_coding_module_typedefs.stamp obj/third_party/webrtc/rtc_base/libbitrate_tracker.a obj/third_party/webrtc/rtc_base/libfrequency_tracker.a obj/third_party/webrtc/rtc_base/one_time_event.stamp obj/third_party/webrtc/rtc_base/librate_limiter.a obj/third_party/webrtc/modules/remote_bitrate_estimator/libremote_bitrate_estimator.a obj/third_party/webrtc/modules/module_api.stamp obj/third_party/webrtc/modules/congestion_controller/goog_cc/liblink_capacity_estimator.a obj/third_party/webrtc/rtc_base/libaudio_format_to_string.a obj/third_party/webrtc/rtc_base/libweak_ptr.a obj/third_party/webrtc/rtc_base/experiments/libbandwidth_quality_scaler_settings.a obj/third_party/webrtc/rtc_base/experiments/libencoder_info_settings.a obj/third_party/webrtc/rtc_base/experiments/libquality_scaler_settings.a obj/third_party/webrtc/rtc_base/experiments/libquality_scaling_experiment.a obj/third_party/webrtc/rtc_base/experiments/librate_control_settings.a obj/third_party/webrtc/rtc_base/experiments/libstable_target_rate_experiment.a obj/third_party/webrtc/modules/video_coding/svc/scalability_mode_util.stamp
  arflags = -T
  output_extension = .a
  output_dir = obj/third_party/webrtc/api/video_codecs
```

reading these is nearly impossible but so far all of them seem to be only in the various tests being built

Reply:
if it's just a ninja dep but nothing is being linked, then it should be fine to just patch out the deps

Reply:
even following things that initially seem more useful like audio_decoder_multiopus end up only visible via various unittests stuff so it seems like clang_rt.builtins (most common one) is only in various tests, etc

but there's no way to just disable this (it's all over the place and i don't think there's any reward for patching this), so it seems whatever

Reply:
biggest benefit from "disabling tests" like this is they're all built for zero reason and make up like 7k+ nodes in the ninja graph, but it's not just 1 toggle and would be patching hundreds of files from what i can see, since it's basically everywhere

Reply:
also seems like all the cargo_crate() stuff adds a dep on compiler_builtins which seems more used

in any case i'm not patching every instance of compiler_builtins/compiler_profile (the only two clang_rt libs used)

Reply:
and for the rust part specifically:  

```
  # Rust and C++ both provide intrinsics for LLVM to call for math operations. We
  # want to use the C++ intrinsics, not the ones in the Rust compiler_builtins
  # library. The Rust symbols are marked as weak, so that they can be replaced by
  # the C++ symbols. This config ensures the C++ symbols exist and are strong in
  # order to cause that replacement to occur by explicitly linking in clang's
  # compiler-rt library.
  if (is_clang && toolchain_has_rust) {
    configs += [ "//build/config/clang:compiler_builtins" ]
  }
```

Reply:
> compiler_profile  

actually that one is gated on `use_clang_coverage` which is always false for us. so it's just the compiler-rt builtins library being explicitly linked for whatever reason. even if we patched it the result would be the same..